compiler/noirc_evaluator/src/ssa/function_builder/mod.rs

```diff
@@ -367,10 +367,12 @@ impl FunctionBuilder {
                 let r_squared = self.insert_binary(r, BinaryOp::Mul, r);
                 let a = self.insert_binary(r_squared, BinaryOp::Mul, lhs);
                 let idx = self.field_constant(FieldElement::from((bit_size - i) as i128));
-                let b = self.insert_array_get(rhs_bits, idx, Type::field());
+                let b = self.insert_array_get(rhs_bits, idx, Type::bool());
```

Comment:
Why are we fetching a bool and casting rather than just fetching a field?

Reply:
We need a bool to be able to `NOT` it efficiently.

Granted this change isn't really necessary, I just think it's more natural to think in terms of "I take a bit from the bit array, not it, then treat these as fields when multiplying" than "I take a field from the bit array, subtract it from 1 then multiply."

Reply:
Ah got it. Your intuition makes more sense to me as well.